File: Projet-LO52/app/src/main/java/fr/utbm/runf1/entities/Race.java
package fr.utbm.runf1.entities;

import java.util.Date;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;
import fr.utbm.runf1.entities.converters.DateConverter;

/**
 * Created by Yosef B.I.
 */
@Entity
@TypeConverters(DateConverter.class)
public class Race {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "race_id")
    private int id;

    private Date date;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
}

File: Projet/app/app/src/main/java/com/codep25/codep25/model/entity/Race.kt
package com.codep25.codep25.model.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(
    tableName = "race"
)
data class Race(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "race_id")
    val id: Long,

    val date: Long,

    @ColumnInfo(name = "nb_turn_per_participant")
    val nbTurnPerParticipant: Int,

    @ColumnInfo(name = "nb_relay")
    val nbOfRelay: Int
)